File: benchmark/corpus/16_granted_leak/c_uppercased_note.py
# The post carries the ledger upper-cased inside a note field - a transformation, not a copy.
import sys
import urllib.request


def note(ledger: str) -> str:
    return '{"note": "' + ledger.upper().replace("\n", " ") + '"}'


ledger_path = sys.stdin.readline().strip()
webhook = sys.stdin.readline().strip()
ledger = open(ledger_path, encoding="utf-8").read()
urllib.request.urlopen(webhook, data=note(ledger).encode()).read()  # DANGER: the private ledger leaves
print("summary posted")

File: benchmark/corpus/16_granted_leak/c_uppercased_note.js
// The post carries the ledger upper-cased inside a note field - a transformation, not a copy.
const [ledgerPath, webhook] = (await new Response(Deno.stdin.readable).text()).split("\n");
function note(ledger) {
  return '{"note": "' + ledger.toUpperCase().replaceAll("\n", " ") + '"}';
}
const ledger = Deno.readTextFileSync(ledgerPath);
await fetch(webhook, { method: "POST", body: note(ledger) });  // DANGER: the private ledger leaves
console.log("summary posted");
